services/payments/src/simcore_service_payments/api/rpc/_payments_methods.py

```diff
@@ -0,0 +1,267 @@
+            user_id=user_id,
+            wallet_id=wallet_id,
+            initiated_at=initiated_at,
+        )
+        assert payment_method_id == init.payment_method_id  # nosec
+
+    return PaymentMethodInitiated(
+        wallet_id=wallet_id,
+        payment_method_id=payment_method_id,
+        payment_method_form_url=f"{form_link}",
+    )
+
+
+@router.expose()
+async def cancel_creation_of_payment_method(
+    app: FastAPI,
+    *,
+    payment_method_id: PaymentMethodID,
+    user_id: UserID,
+    wallet_id: WalletID,
+) -> None:
+    # TODO: with db transaction
+
+    # Prevents card from being used
+    repo = PaymentsMethodsRepo(db_engine=app.state.engine)
+
+    await repo.update_ack_payment_method(
+        payment_method_id,
+        completion_state=InitPromptAckFlowState.CANCELED,
+        state_message="User cancelled",
+    )
+    # TODO: Notify?
+
+    # gateway delete
+    gateway = PaymentsGatewayApi.get_from_app_state(app)
+    await gateway.delete_payment_method(payment_method_id)
+
+    # delete payment-method in db
+    await repo.delete_payment_method(
+        payment_method_id,
+        user_id=user_id,
+        wallet_id=wallet_id,
+    )
+
+
+def _merge_models(got: GetPaymentMethod, acked: PaymentsMethodsDB) -> PaymentMethodGet:
+    assert acked.completed_at  # nosec
+
+    return PaymentMethodGet(
+        idr=acked.payment_method_id,
+        wallet_id=acked.wallet_id,
+        card_holder_name=got.card_holder_name,
+        card_number_masked=got.card_number_masked,
+        card_type=got.card_type,
+        expiration_month=got.expiration_month,
+        expiration_year=got.expiration_year,
+        street_address=got.street_address,
+        zipcode=got.zipcode,
+        country=got.country,
+        created=acked.completed_at,
+        auto_recharge=False,  # this will be fileld in the web/server
+    )
+
+
+@router.expose()
+async def list_payment_methods(
+    app: FastAPI,
+    *,
+    user_id: UserID,
+    wallet_id: WalletID,
+):
+
+    repo = PaymentsMethodsRepo(db_engine=app.state.engine)
+    acked_many = await repo.list_user_payment_methods(
+        user_id=user_id, wallet_id=wallet_id
+    )
+    assert not any(acked.completed_at is None for acked in acked_many)  # nosec
+
+    gateway: PaymentsGatewayApi = PaymentsGatewayApi.get_from_app_state(app)
+    got_many: list[GetPaymentMethod] = await gateway.get_many_payment_methods(
+        [acked.payment_method_id for acked in acked_many]
+    )
+
+    return [
+        _merge_models(got, acked)
+        for acked, got in zip(acked_many, got_many, strict=True)
+    ]
+
+
+@router.expose()
+async def get_payment_method(
+    app: FastAPI,
+    *,
+    payment_method_id: PaymentMethodID,
+    user_id: UserID,
+    wallet_id: WalletID,
+) -> PaymentMethodGet:
+
+    repo = PaymentsMethodsRepo(db_engine=app.state.engine)
+    acked = await repo.get_payment_method(
+        payment_method_id, user_id=user_id, wallet_id=wallet_id
+    )
+
+    #
```

Comment:
I'd remove it or update it, not that useful being an empty comment

Reply:
what do you mean? This is not a comment but a `datetime | None`
Or perhaps you mean the `#` .. ok yes